bitmark-references/bitmark-node-software/bitmark-blockchain.md

```diff
@@ -0,0 +1,173 @@
+  depend on the type code inside the Account field.
+
+
+## Transactions: Asset, Issue and Transfer Records
+
+### Asset Record
+
+This contains the metadata to describe an asset that will be issued on
+the blockchain.  All assets will have at least one issue.
+
+The Asset Record structure
+
+Item              Type       Description
+----------------  ---------  ------------------------
+Type code         Byte       Asset Record type code
+Name              String     Short descriptive name
+Fingerprint       String     A unique identifier for the Asset
+Metadata          Map        Key-Value to describe Asset
+Registrant        Account    The public key of the signer
+Signature         Signature  Ed25519 signature of signer
+
+
+The Issue Record structure
+
+Item              Type       Description
+----------------  ---------  ------------------------
+Type code         Byte       Issue Record type code
+AssetId           Binary     Link to asset record (SHA3-512 of asset fingerprint)
+Owner             Account    The public key of the signer
+Nonce             VarInt     Allow for multiple issues of the same asset
+Signature         Signature  Ed25519 signature of signer
+
+The Transfer Record structure
+
+Item              Type       Description
+----------------  ---------  ------------------------
+Type code         Byte       Transfer Record type code
+Link              Binary     Link to previous Issue or Transfer (SHA3-256)
+Owner             Account    The public key of the signer
+Signature         Signature  Ed25519 signature of linked previous owner
+CounterSignature  Signature  Ed25519 signature of this record owner
+
+
+## Transactions: Share
+
+
+## Transactions: Block ownership record
+
+
+## Blockchain structure
+
+### Block Header
+
+Item              Type       Description
+----------------  ---------  ------------------------
+Version           U16        The version control block rules in action
+TransactionCount  U16        Number of transactions in the block
+Number            U64        The number of this block
+PreviousBlock     Argon2     Thirty two byte hash of the previous block
+MerkleRoot        SHA3-256   Thirty two byte hash of the transaction Merkle tree
+Timestamp         U64        Timestamp as a Unix (seconds after start of 1970-01-01)
+Difficulty        U64        Difficulty fraction as 57 bit mantissa + 8 bit exponent
+Nonce             U64        Nonce created by hashing to meet the difficulty
+
+### Merkle tree
+
+This is a way of combining hashes of a list of records to obtain a
+single hash such that only the same records in the same order will
+have the same hash.  It is done by making the hashes of all the
+records the final elements in a binary tree, with each pair hashed
+together to get the value above.  Any item not forming a pair at
+some level of the tree is hashed with itself.  The final hash is known
+as the root.  In this system the SHA3-256 algorithm is used for the
+hashing process.
+
```

Comment:
This description is confusing. Maybe something like:

A Merkle Tree combines the hashes of individual records, such that only the same records hashed together in the same order will result in the same Merkle Tree root. It does so by hashing individual transactions to create the leaves of a binary tree. Each pair of hashes in the tree is then combined to form a new hash, repeating the process until only a single hash is left: the Merkle Tree root. Whenever a hash in the tree does not have a paired hash to combine with, it is instead hashed with itself. The hashing process is conducted by the SHA3-256 algorithm.